File: app/src/main/java/com/diu/yk_games/line2box/NameInfoFragment.java
package com.diu.yk_games.line2box;

import android.content.Context;
import android.content.SharedPreferences;
import android.media.MediaPlayer;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;

import java.util.Objects;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link NameInfoFragment# newInstance} factory method to
 * create an instance of this fragment.
 */
public class NameInfoFragment extends Fragment
{
    EditText nm1EditText, nm2EditText ;
    public String nm1="Red", nm2="Blue",tNm1,tNm2;
    boolean nm1Def=true;

    @Override
    public void onDetach()
    {
        super.onDetach();
        ((GameActivity1) requireActivity()).onStopFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater,
                             ViewGroup container,
                             Bundle savedInstanceState)
    {
        // Inflate the layout for this fragment
        View view= inflater.inflate(R.layout.fragment_nminfo, container, false);
        SharedPreferences sharedPref = requireContext().getSharedPreferences(
                getString(R.string.preference_file_key), Context.MODE_PRIVATE);
        SharedPreferences.Editor prefEditor = sharedPref.edit();
        GameProfile.setPreferences(sharedPref);

        CheckBox nmSave=view.findViewById(R.id.nmSaveBox);
        nm1EditText=view.findViewById(R.id.palyerRed);
        nm2EditText=view.findViewById(R.id.palyerBlue);

        //nm1EditText.setEnabled(false);

        String tmpNm1= sharedPref.getString("plrNm1",""),
                tmpNm2= sharedPref.getString("plrNm2",""),
                prfNm=new GameProfile().nm;
        if(!tmpNm1.equals(""))
            nm1EditText.setText(tmpNm1);
        else
            nm1EditText.setText(prfNm);
        if(!tmpNm2.equals(""))
            nm2EditText.setText(tmpNm2);

        //nmEditText.setSelection(nmEditText.getText().length());

        //nm1=new GameProfile().nm;

        //new ArrayAdapter<String>()

        Button btn = view.findViewById(R.id.playBtn);
        btn.setOnClickListener(arg0 ->
        {
            if(!sharedPref.getBoolean("muted", false))
            {
                MediaPlayer mediaPlayer = MediaPlayer.create(getContext(), R.raw.btn_click_ef);
                mediaPlayer.start();
                mediaPlayer.setOnCompletionListener(MediaPlayer::release);
            }
            if(!nm1EditText.getText().toString().equals(""))
            {
                nm1=nm1EditText.getText().toString();
            }
            if(!nm2EditText.getText().toString().equals(""))
            {
                nm2=nm2EditText.getText().toString();
            }
            if(nmSave.isChecked())
            {
                if(!nm1EditText.getText().toString().equals(""))
                    prefEditor.putString("plrNm1",nm1).apply();
                if(!nm2EditText.getText().toString().equals(""))
                    prefEditor.putString("plrNm2",nm2).apply();
            }
            GameActivity1.nm1=nm1;
            GameActivity1.nm2=nm2;
            requireActivity().getSupportFragmentManager().beginTransaction().remove(this).commit();
            //String Item = getActivity().getIntent().getExtras().getString("name");
            //getContext().getView().findViewById(R.id.nmLayout).setVisibility(View.VISIBLE);


        });
        view.findViewById(R.id.nmSwanBtn).setOnClickListener(x ->
        {
            if(!sharedPref.getBoolean("muted", false))
            {
                MediaPlayer mediaPlayer = MediaPlayer.create(getContext(), R.raw.btn_click_ef);
                mediaPlayer.start();
                mediaPlayer.setOnCompletionListener(MediaPlayer::release);
            }
            tNm1=nm1EditText.getText().toString();
            tNm2=nm2EditText.getText().toString();
            nm1EditText.setText(tNm2);
            nm2EditText.setText(tNm1);

        });
        return view;
    }


}

File: app/src/main/java/com/diu/yk_games/line2box/presentation/offline/NameInfoFragment.kt
package com.diu.yk_games.line2box.presentation.offline

import android.os.Bundle
import android.view.View
import androidx.lifecycle.lifecycleScope
import com.diu.yk_games.line2box.databinding.FragmentNminfoBinding
import com.diu.yk_games.line2box.model.GameProfile
import com.diu.yk_games.line2box.presentation.base.BaseFragment
import com.diu.yk_games.line2box.presentation.main.SettingsFragment
import com.diu.yk_games.line2box.presentation.navigation.Routes
import com.diu.yk_games.line2box.util.*
import kotlinx.coroutines.launch

class NameInfoFragment : BaseFragment<FragmentNminfoBinding>(FragmentNminfoBinding::inflate) {
    private var nm1 = "Red"
    private var nm2 = "Blue"

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupUI()
        setupListener()
    }

    private fun setupUI() {
        lifecycleScope.launch {
            binding.apply {
                val tmpNm1 = IO { pref.read("plrNm1", "") }
                val tmpNm2 = IO { pref.read("plrNm2", "") }
                val prfNm = IO { GameProfile().nm }
                palyerRed.setText(if (tmpNm1 != "") tmpNm1 else prfNm)
                if (tmpNm2 != "") palyerBlue.setText(tmpNm2)
            }
        }
    }

    fun setupListener() {
        binding.settingBtn.setBounceClickListener {
            viewModel.player.playButtonClickSound()
            SettingsFragment.show(childFragmentManager)
        }
        binding.backBtn.setBounceClickListener {
            viewModel.player.playButtonClickSound()
            onBackPressed()
        }
        binding.playBtn.setBounceClickListener {
            viewModel.player.playButtonClickSound()
            if (binding.palyerRed.text.toString() != "") nm1 = binding.palyerRed.text.toString()

            if (binding.palyerBlue.text.toString() != "") nm2 = binding.palyerBlue.text.toString()

            if (binding.nmSaveBox.isChecked) {
                if (binding.palyerRed.text.toString() != "")
                    pref.save("plrNm1", nm1)
                if (binding.palyerBlue.text.toString() != "")
                    pref.save("plrNm2", nm2)
            }
            navigateSafe(Routes.GameDual(nm1.trim(), nm2.trim())) {
                popUpTo(Routes.ChangeName::class){
                    inclusive = true
                }
            }
        }
        binding.nmSwanBtn.setBounceClickListener {
            viewModel.player.playButtonClickSound()
            binding.palyerRed.text.toString().let {
                binding.palyerRed.setText(binding.palyerBlue.text.toString())
                binding.palyerBlue.setText(it)
            }
        }
    }
}